src/video/x11/SDL_x11xinput2.c

```diff
@@ -144,6 +158,59 @@ X11_InitXinput2(_THIS)
     }
 
     xinput2_initialized = 1;
+    xinput2_precise_scroll_supported = xinput2_version_atleast(version, 2, 1);
+
+    SDL_memset(scrollable_devices, 0, sizeof(scrollable_devices));
+    for (i = 0; i<MAX_SCROLLABLE_DEVICES; i++) {
+        scrollable_devices[i].source_id = -1;
+    }
+
+    if (xinput2_precise_scroll_supported) {
+        XIDeviceInfo *info;
+        int ndevices,i,j,k,dev_i=0;
+
+        info = X11_XIQueryDevice(data->display, XIAllDevices, &ndevices);
+
+        /* find scroll classes */
+        for (i = 0; i < ndevices && dev_i < MAX_SCROLLABLE_DEVICES; i++) {
+            scroll_device *sd = scrollable_devices[dev_i];
+            XIDeviceInfo *dev = &info[i];
+            for (j = 0; j < dev->num_classes; j++) {
+                XIAnyClassInfo *class = dev->classes[j];
+                XIScrollClassInfo *s = (XIScrollClassInfo*)class;
+
+                if (class->type != XIScrollClass)
+                    continue;
+
+                int dir = s->scroll_type == XIScrollTypeVertical;
```

Comment:
This will hit `-Werror=declaration-after-statement`

Reply:
https://github.com/libsdl-org/SDL/runs/5441254825?check_suite_focus=true#step:6:189

Reply:
Should be fixed in ee4fa6e, along with a typo'd type name I somehow managed to mess up between testing and committing...